### PR TITLE
LDK 0.0.104

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Built on #41, this updates us to LDK 0.0.104!

### CODE REVIEW

**Review by jkczyz — COMMENTED** [review]
Could you update `keysend` to use `InvoicePayer::pay_pubkey`?

**Comment by TheBlueMatt** [comment]
> Could you update keysend to use InvoicePayer::pay_pubkey?

Done

**Review by jkczyz — APPROVED** [review]

File: src/cli.rs [hunk 1]
```diff
@@ -592,40 +585,47 @@ fn send_payment<E: EventHandler>(
...
 	) {
-		Ok(r) => r,
-		Err(e) => {
+		Ok(_payment_id) => {
+			println!("EVENT: initiated sending {} msats to {}", amt_msat, payee_pubkey);
+			print!("> ");
+			HTLCStatus::Pending
+		}
+		Err(PaymentError::Invoice(e)) => {
+			println!("ERROR: invalid payee: {}", e);
```
[the remark this answers is not in the document]
Comment (TheBlueMatt) [reply src/cli.rs:606]: Yea, I kinda figured it didn't matter so I didn't bother thinking hard cause it *should* be unreachable.

File: src/cli.rs [hunk 2]
```diff
@@ -592,40 +585,47 @@ fn send_payment<E: EventHandler>(
...
-		&params,
-		&network_graph,
-		Some(&first_hops.iter().collect::<Vec<_>>()),
-		logger,
-		&scorer.lock().unwrap(),
+	let payment_preimage = keys.get_secure_random_bytes();
+
+	let status = match invoice_payer.pay_pubkey(
+		payee_pubkey,
+		PaymentPreimage(payment_preimage),
```
[the remark this answers is not in the document]
Comment (TheBlueMatt) [reply src/cli.rs:596]: Yea, taking an `Option<PaymentPreimage>` definitely makes sense IMO, returning `PaymentHash` I guess would be nice too.

**Comment by TheBlueMatt** [comment]
Oops this should have landed a while ago, I guess the comments weren't blocking?

**Pull request merged**